CONTRIBUTING.md

```diff
@@ -1,57 +1,295 @@
 
-### How to Use `/* prettier-ignore */`
+### Installation
 
-To use `/* prettier-ignore */`, simply place the comment on the line before the code you want to ignore. Here is an example:
+1. **Clone the Repository**:
+    ```bash
+    git clone https://github.com/shardeum/shardeum.git
+    cd shardeum
+    ```
 
-#### Example
+2. **Install Dependencies**:
+    ```bash
+    npm install
+    ```
 
-```javascript
-/* prettier-ignore */
-if (ShardeumFlags.VerboseLogs) console.log('Lengths of blocks after pruning', Object.keys(blocksByHash).length, Object.keys(readableBlocks).length)
+3. **Setup Environment Variables**:
+    Copy the `.env_example` to `.env` and customize the settings as needed.
+    ```bash
+    cp .env_example .env
+    ```
+
+### Node.js and NVM
+
+Shardeum requires a specific version of Node.js (18.16.1) and npm (9.5.1). Use the Node Version Manager (NVM) to manage multiple versions of Node.js:
+```bash
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+nvm install 18.16.1
+nvm use 18.16.1
 ```
+
+### Rust Toolchain
+
+Some networking code is implemented in Rust. Install Rust as follows:
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+rustup install 1.74.1
+rustup default 1.74.1
+```
+
+### Build Essentials
+
+Install necessary build tools:
+- For Linux:
+    ```bash
+    sudo apt-get install build-essential
+    ```
+- For MacOS:
+    ```bash
+    brew install gcc
+    ```
+
+### Node-gyp
+
+Install node-gyp globally and configure Python:
+```bash
+npm i -g node-gyp
+npm config set python `which python3`
+npm config list
+```
+
+## Using ESLint and Prettier
+
+Ensure consistent code style using ESLint and Prettier. We use the following configuration:
+- **ESLint**: For JavaScript and TypeScript linting.
+- **Prettier**: For code formatting.
+
+### Install ESLint and Prettier
+
+1. **Add ESLint to the project**:
+    ```bash
+    npm install eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin --save-dev
+    ```
+
+2. **Create or update `.eslintrc.json`**:
+    ```json
+    {
+      "parser": "@typescript-eslint/parser",
+      "plugins": ["@typescript-eslint"],
+      "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+      "rules": {
+        "indent": ["error", 4],
+        "@typescript-eslint/indent": ["error", 4]
+      }
+    }
+    ```
+
+3. **Install Prettier and configure**:
+    ```bash
+    npm install prettier eslint-config-prettier eslint-plugin-prettier --save-dev
+    ```
+
+4. **Create or update `.prettierrc`**:
+    ```json
+    {
+      "singleQuote": true,
+      "semi": false,
+      "tabWidth": 4,
+      "useTabs": false
+    }
+    ```
+
+5. **Install ESLint extension for VS Code** for linting and auto-fixing on save.
+
+## Continuous Integration and Deployment
```

Comment:
We can remove this section since we no longer use Gitlab. We've migrated our codebases to Github, we'll remove any gitlab yml files lingering in the codebase